prov/sockets/src/sock_progress.c

```diff
@@ -2626,13 +2627,18 @@ static void sock_pe_poll(struct sock_pe *pe)
 	struct sock_tx_ctx *tx_ctx;
 	struct sock_rx_ctx *rx_ctx;
 
-	if (pe->waittime && ((fi_gettime_ms() - pe->waittime) < sock_pe_waittime))
+	pthread_mutex_lock(&pe->list_lock);
+	if (pe->waittime && ((fi_gettime_ms() - pe->waittime) <
+				sock_pe_waittime)) {
+		pthread_mutex_unlock(&pe->list_lock);
 		return;
+	}
 
-	if (dlist_empty(&pe->tx_list) && dlist_empty(&pe->rx_list))
+	if (dlist_empty(&pe->tx_list) && dlist_empty(&pe->rx_list)) {
+		pthread_mutex_unlock(&pe->list_lock);
 		return;
+	}
```

Comment:
The above changes look like it leaves the mutex locked if neither of the if statements are followed.  Can we instead use a goto to jump to a common unlock call that will always be called?